src/openms/include/OpenMS/ANALYSIS/ID/PeptideIndexing.h

```diff
@@ -328,6 +329,14 @@ namespace OpenMS
               String prot = proteins.chunkAt(i).sequence;
 
               prot.remove('*');
+
+              // check for invalid sequences with modifications
+              if (prot.has('[') || prot.has('('))
+              { 
+                 invalid_protein_sequence = true; // not omp-critical because its write-only
+                 prot = AASequence::fromString(prot).toUnmodifiedString();
```

Comment:
I think this is not thread-safe (and can't easily be fixed (without huge performance penalties) because ResidueDB is not thread-safe).
I am pretty sure that this might lead to a concurrent read/write to ResidueDB.

Reply:
see https://github.com/OpenMS/OpenMS/issues/905

Reply:
please correct me if I am wrong

Reply:
you're right. So I'm going to elevate to error for now until we have a solution for ResidueDB.